src/components/rangeUsePlan/RangeUsePlan.js

```diff
@@ -28,52 +28,54 @@ const defaultProps = {
 };
 
 export class RangeUsePlan extends Component {
-  state = {
-    isCompletedModalOpen: false,
-    isPendingModalOpen: false,
-    isUpdateZoneModalOpen: false,
-    zone: {},
-    status: {},
-  }
+  constructor(props) {
+    super(props);
 
-  componentDidMount() {
     // store fields that can be updated within this page
-    const { zone, plans } = this.props.agreement;
+    const { zone, plans } = props.agreement;
     const plan = plans[0];
     const status = plan && plan.status;
-    this.setState({
+
+    this.state = {
+      isCompletedModalOpen: false,
+      isPendingModalOpen: false,
+      isUpdateZoneModalOpen: false,
       zone,
       status,
       plan,
-    });
+    };
   }
 
   onViewClicked = () => {
     this.pdfLink.click();
   }
 
-  openCompletedConfirmModal = () => this.setState({ isCompletedModalOpen: true })
-
-  closeCompletedConfirmModal = () => this.setState({ isCompletedModalOpen: false })
+  onYesCompletedClicked = () => {
+    this.updateStatus(COMPLETED, this.closeCompletedConfirmModal);
+  }
 
-  openPendingConfirmModal = () => this.setState({ isPendingModalOpen: true })
-  
-  closePendingConfirmModal = () => this.setState({ isPendingModalOpen: false })
+  onYesPendingClicked = () => {
+    this.updateStatus(PENDING, this.closePendingConfirmModal);
+  }
 
-  openUpdateZoneModal = () => this.setState({ isUpdateZoneModalOpen: true })
+  onZoneClicked = () => {
+    this.openUpdateZoneModal();
```

Comment:
might be nice to make the modal function more generic - but it's definitely fine as it is too! You also might want to think about keeping modal state in Redux - might not be necessary now but going forward it might save you a lot of trouble - something to keep in mind.